src/lib/sparse-buffer.ts

```diff
@@ -319,7 +325,38 @@ export class SparseBuffer implements ISparseBuffer {
             undefined
         );
     }
+}
+
+function trim(result: INumberArray) {
+    result.length = result.count;
+    (result as any).count = undefined;
+}
 
+export function shuffle(numbers: number[]): void {
+    // fisher-yates shuffle algorithm
+    for (let i = numbers.length - 1; i > 0; i--) {
+        const
+            j = randomNumber(0, i),
+            swap = numbers[i];
+        numbers[i] = numbers[j];
+        numbers[j] = swap;
+    }
+}
+
+export type VoidFunc = ((...args: any[]) => void);
+
+export function repeat(
+    howManyTimes: number,
+    activity: VoidFunc) {
+    for (let i = 0; i < howManyTimes; i++) {
+        activity.call(null, [ i ]);
+    }
+}
+
+function randomNumber(min: number, max: number) {
+    return Math.round(
```

Comment:
I think this should be Math.floor - I assume it's possible to hit the max of the random param, which according to Fisher Yates is a no-no because you may choose the item you are trying to shuffle?

Reply:
according to https://www.geeksforgeeks.org/shuffle-a-given-array-using-fisher-yates-shuffle-algorithm/ (where I picked up the idea) ` 0 <= j <= i`, in which case, the upper bound is acceptable.